zebra-network/src/peer/handshake.rs

```diff
@@ -1017,6 +959,78 @@ where
     }
 }
 
+/// Send periodical heartbeats to `server_tx`, and update the peer status through
+/// `heartbeat_ts_collector`.
+///
+/// # Correctness
+///
+/// To prevent hangs:
+/// - every await that depends on the network must have a timeout (or interval)
+/// - every error/shutdown must update the address book state and return
+///
+/// The address book state can be updated via `ClientRequest.tx`, or the
+/// heartbeat_ts_collector.
+///
+/// Returning from the spawned closure terminates the connection's heartbeat task.
```

Comment:
Nit: not a closure any more

```suggestion
/// Returning from this function terminates the connection's heartbeat task.
```

Reply:
Done.